### PR TITLE
Add TLS protocol

### PR DESCRIPTION
**Pull request by carpawell**
Added TLS protocol. There are plenty of applications of TLS protocol not in the context of using it with `http` or `ws`(e.g. in `gPRC`), so it seems logical to add it. Code for it was taken from [multicodec](https://github.com/multiformats/multicodec) repo.

There is https://github.com/multiformats/multiaddr/pull/109 PR that suggests adding TLS protocol, mark `https` and `wss` protocols as deprecated and start using `/tls/http` and `/tls/ws` instead.

### CODE REVIEW

File: multiaddr_test.go
Comment (Stebalien) [inline multiaddr_test.go:143]:
```suggestion
		"/tcp/1234/tls/http",
```

File: multiaddr_test.go
Comment (Stebalien) [inline multiaddr_test.go:172]:
```suggestion
		"/ip4/127.0.0.1/tcp/127/tls/ws",
```

**Review by Stebalien — APPROVED** [review]
As long as we're not simply _removing_ /https, this seems reasonable. Let me quickly get a signoff from other libp2p devs but then I think we can probably just merge it.

**Review by mxinden — APPROVED** [review]
This looks good to me from the multiformats perspective. I can not comment on the Go side.

I think we should wait for a sign-off by @marten-seemann as well to ensure this is consistent long term with QUIC and HTTP3.

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**
